src/Microsoft.PowerShell.Commands.Management/commands/management/Process.cs

```diff
@@ -640,6 +640,10 @@ protected override void ProcessRecord()
                             WriteNonTerminatingError(process, ex, ProcessResources.CouldNotEnumerateModules, "CouldNotEnumerateModules", ErrorCategory.PermissionDenied);
                         }
                     }
+                    catch (PipelineStoppedException) 
+                    {
+                        throw;
+                    }
```

Comment:
I guess we will see the same for any collection. If so would the fix be in WriteObject() method?

Reply:
I think that since we have to catch errors from WriteObject() here anyway and make them non-terminating so we can enumerate a process's modules, "fixing" this in WriteObject() itself would just give us the same issue in this cmdlet.

I'm almost positive that rethrowing a PipelineStoppedException is something WriteObject() is already doing here.

Reply:
Yeah I'm not sure of a better way to do this, catching `PipelineStoppedException` then rethrowing will stop the whole pipeline and avoid writing the non-terminating error. 

Seems like other places in the code doing this well to handle Select-Object:

https://github.com/PowerShell/PowerShell/blob/658837323599ab1c7a81fe66fcd43f7420e4402b/src/System.Management.Automation/engine/InternalCommands.cs#L770-L774

Reply:
Uh, there doesn't seem to be a common solution.